src/graphdb.js

```diff
@@ -52,20 +52,66 @@ function toPascalCase (str) {
     return r.trim();
 }
 
+// Changes every instance of invalid characters in the given label with the following delimiters
+function replaceCleanseLabel(label) {
+    const delimiterExclamation = "_exclamationmark_";
+    const delimiterDollar = "_dollarsign_";
+    const delimiterAmpersand = "_ampersand_";
+    const delimiterOpenBracket = "_openbracket_";
+    const delimiterCloseBracket = "_closebracket_";
+    const delimiterPeriod = "_period_";
+    const delimiterColon = "_colon_";
+    const delimiterEqual = "_equal_";
+    const delimiterAt = "_at_";
+    const delimiterOpenSquareBracket = "_opensquarebracket_";
+    const delimiterCloseSquareBracket = "_closesquarebracket_";
+    const delimiterOpenCurlyBracket = "_opencurlybracket_";
+    const delimiterVerticalBar = "_verticalbar_";
+    const delimiterCloseCurlyBracket = "_closecurlybracket_";
+    const delimiterHyphen = "_hyphen_";
+
+    return label
+        .replaceAll("!", delimiterExclamation)
```

Comment:
Suggest to inline the replacements so that it's easier to see what is replacing each character and also make the replacements shorter abbreviations:

```
function replaceCleanseLabel(label) {
    return label
        .replaceAll("!", "_ex_")
        .replaceAll("$", "_dol_")
        .replaceAll("&", "_amp_")
        .replaceAll("(", "_op_")
        .replaceAll(")", "_cp_")
        .replaceAll(".", "_dot_")
        .replaceAll(":", "_cn_")
        .replaceAll("=", "_eq_")
        .replaceAll("@", "_at_")
        .replaceAll("[", "_os_")
        .replaceAll("]", "_cs_")
        .replaceAll("{", "_oc_")
        .replaceAll("|", "_vb_")
        .replaceAll("}", "_cc_")
        .replaceAll("-", "_hy_");
}
```